Simple Login Screen/LoginViewController.swift

```diff
@@ -30,14 +30,125 @@ class LoginViewController: UIViewController {
 
     
     @IBAction func signupButtonPressed(_ sender: Any) {
+        
+        //Clear keyboard
+        self.view.endEditing(true)
+        
+        //Add values to the arrays
         emailArray.append(emailTextField!.text!)
         usernameArray.append(usernameTextField!.text!)
         passwordArray.append(passwordTextField!.text!)
+        
+        //Clear text fields
+        emailTextField.text = ""
+        usernameTextField.text = ""
+        passwordTextField.text = ""
+    }
+    
+    //Tells if email is good
+    @IBAction func emailEditingDidEnd(_ sender: Any) {
+        
+        if validateEmailExisting(text: emailTextField.text!) && validateEmailFormat(candidate: emailTextField.text!)  {
+            
+            print ("Email good.")
+            
+        } else {
+            
+            print("Email bad")
+        }
+        
+    }
+
+    
+    
+    //Tells if username is long enough
+    @IBAction func usernameEditingDidEnd(_ sender: Any) {
+        
+        if validateUsernameLength(text: usernameTextField.text!) {
+            
+            print("Username good")
+            
+        } else {
+            
+            print("Username bad")
+        }
+    }
+    
+    //Tells if password is good
+    @IBAction func passwordEditingDidEnd(_ sender: Any) {
+        
+        if validatePasswordFormat(candidate: passwordTextField.text!) {
+            
+            print("password good")
+            
+        } else {
+            
+            print("password bad")
+            
+        }
+    }
+    
+    
+    //Email Validation Existing
+    func validateEmailExisting(text: String) -> Bool {
+        
+        var result = true
+        
+        if emailArray.contains(emailTextField.text!) {
+            
+            result = false
+        }
+        
+        return result
+        
+    }
+    
+    //Email Validation Format. True means email is good.
+    func validateEmailFormat(candidate: String) -> Bool {
+        
+        let emailRegex = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}"
+        
+        return NSPredicate(format: "SELF MATCHES %@", emailRegex).evaluate(with: candidate)
+    }
+    
+    //Username Validation Existing. True means not existing.
+    func validateUsernameExisting(text: String) -> Bool {
+        
+        var result = true
+        
+        if usernameArray.contains(usernameTextField.text!) {
+            
+            result = false
+        }
+        
+        return result
+        
     }
     
+    //Username Validation Length. True means username is long enough.
+    func validateUsernameLength(text: String) -> Bool {
+        
+        var result = true
+        
+        if ((usernameTextField.text?.characters.count)! < 9) {
```

Comment:
this statement returns the boolean you need, the rest of this method is redundant.